internal/pkg/template/diff/diff.go

```diff
@@ -90,7 +91,75 @@ func isYAMLLeaf(node *yaml.Node) bool {
 }
 
 func parseSequence(from, to *yaml.Node) (map[string]*Node, error) {
-	return nil, nil
+	fromSeq, toSeq := make([]yaml.Node, len(from.Content)), make([]yaml.Node, len(to.Content))
+	if err := to.Decode(&toSeq); err != nil {
+		return nil, err
+	}
+	if err := from.Decode(&fromSeq); err != nil {
+		return nil, err
+	}
+	cachedDiff, cachedErr := make(map[string]*Node), make(map[string]error)
+	lcsIdxInFrom, lcsIdxInTo := longestCommonSubsequence(fromSeq, toSeq, func(inA, inB int) bool {
+		diff, err := parse(&(fromSeq[inA]), &(toSeq[inB]), "")
+		if diff != nil { // NOTE: cache the diff only if a modification could have happened at this position.
+			cachedDiff[cachedKey(inA, inB)] = diff
+			cachedErr[cachedKey(inA, inB)] = err
+		}
+		return err == nil && diff == nil // NOTE: we swallow the error for now.
+	})
+	childKey := seqChildKeyFunc()
+	children := make(map[string]*Node)
+	var f, t, i int
+	for {
+		if i >= len(lcsIdxInFrom) {
+			break
+		}
+		lcsF, lcsT := lcsIdxInFrom[i], lcsIdxInTo[i]
+		switch {
+		case f == lcsF && t == lcsT: // Match.
```

Comment:
nit: I wonder if this body would be easier to grok if things were laid out like:
| current name | proposed |
|:--------|:------|
| `f` | `fromIdx` |
| `t` | `toIdx` |
| `i` | `lcsIdx` |


and we change the function signature of:
```go
// before
longestCommonSubsequence[T any](a []T, b []T, eq eqFunc) ([]int, []int)

// after
type lcsIndex struct {
  from int
  to int
}
longestCommonSubsequence[T any](a []T, b []T, eq eqFunc) []lcsIndex
```

then the whole thing can read:
```go
lcs := longestCommonSubsequence(fromSeq, toSeq, ...)

// other stuff...

lcsIdx := lcs[lcsIndex]
switch {
case fromIdx == lcsIdx.from && toIdx == lcsIdx.to: // and so on...
}
```

Just a thought 💭 I'd be curious to hear if others find that easier to follow or if they have a different suggestion.

Reply:
Good idea 👍🏼 lmk how you like the current `iter` implementation & if it smells like over-engineering 😂